src/index.ts

```diff
@@ -139,6 +141,27 @@ class MetricsGatherer {
 		};
 	}
 
+	aggregateRequestHandler(authTest? : AuthTestFunc, callback? : Function) : express.Handler {
+		const aggregatorRegistry = new prometheus.AggregatorRegistry();
+		return (req, res) => {
+			if (authTest && !authTest(req)) {
+				res.status(403).send();
+			}
+			aggregatorRegistry.clusterMetrics()
+				.then((metrics: string) => {
+					res.set('Content-Type', aggregatorRegistry.contentType);
+					res.send(metrics);
+					if (callback) {
+						callback();
+					}
+				})
+				.catch((err: Error) => {
+					console.error(`error in /cluster_metrics: ${err}`);
```

Comment:
Can you remove this or put it behind a debug flag, having modules log directly/unconditionally removes control from the dependants

Reply:
I think uncaught errors in promise chains will start to crash the runtime (or do they already do this? I know in 9.x it would error-log, but I kept reading it would start to crash it in a future version), so I'll put it behind a flag.

Reply:
The `.catch` is what avoids the uncaught error issue, logging or not is unrelated to that.  As for it crashing the process in the future, I haven't heard anything about that but it's possible - either way it won't affect us here though as we do handle with the `res.send`